tests/integration/test_types_integration.py

```diff
@@ -142,6 +145,166 @@ def test_json(trino_connection):
         .execute()
 
 
+def test_date(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS DATE)", python=None) \
+        .add_field(sql="DATE '2001-08-22'", python=date(2001, 8, 22)) \
+        .add_field(sql="DATE '0001-01-01'", python=date(1, 1, 1)) \
+        .add_field(sql="DATE '1582-10-04'", python=date(1582, 10, 4)) \
+        .add_field(sql="DATE '1582-10-05'", python=date(1582, 10, 5)) \
+        .add_field(sql="DATE '1582-10-14'", python=date(1582, 10, 14)) \
+        .execute()
+
+
+def test_time(trino_connection):
+    time_0 = time(1, 23, 45)
+    time_3 = time(1, 23, 45, 123000)
+    # Because of `PARAMETRIC_DATETIME` not being set, values are rounded to milliseconds
+    time_6 = time(1, 23, 45, 123000)
+    time_round_down = time(1, 23, 45, 123000)
+    time_round_up = time(1, 23, 45, 124000)
+    time_round_up_second = time(1, 23, 46)
+    time_leading_0 = (datetime(2, 1, 1, 11, 23, 45)).time()
+
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS TIME)", python=None) \
+        .add_field(sql="TIME '01:23:45'", python=time_0) \
+        .add_field(sql="TIME '01:23:45.123'", python=time_3) \
+        .add_field(sql="TIME '01:23:45.123'", python=time_3) \
+        .add_field(sql="TIME '01:23:45.123456'", python=time_6) \
+        .add_field(sql="TIME '01:23:45.123456789'", python=time_round_down) \
+        .add_field(sql="TIME '01:23:45.123456789123'", python=time_round_down) \
+        .add_field(sql="TIME '01:23:45.123900'", python=time_round_up) \
+        .add_field(sql="TIME '01:23:45.999900'", python=time_round_up_second) \
+        .add_field(sql="TIME '11:23:45.000100'", python=time_leading_0) \
+        .execute()
```

Comment:
```suggestion
    (
        SqlTest(trino_connection)
            .add_field(sql="CAST(null AS TIME)", python=None)
            .add_field(sql="TIME '00:00:00'", python=time(0, 0, 0)) # min supported time(3)
            .add_field(sql="TIME '12:34:56.123'", python=time(12, 34, 56, 123000))
            .add_field(sql="TIME '23:59:59.999'", python=time(23, 59, 59, 999000)) # max supported time(3)
            # min value for each precision
            .add_field(sql="TIME '00:00:00'", python=time(0, 0, 0))
            .add_field(sql="TIME '00:00:00.1'", python=time(0, 0, 0, 100000))
            .add_field(sql="TIME '00:00:00.01'", python=time(0, 0, 0, 10000))
            .add_field(sql="TIME '00:00:00.001'", python=time(0, 0, 0, 1000))
            .add_field(sql="TIME '00:00:00.0001'", python=time(0, 0, 0, 100))
            .add_field(sql="TIME '00:00:00.00001'", python=time(0, 0, 0, 10))
            .add_field(sql="TIME '00:00:00.000001'", python=time(0, 0, 0, 1))
            # max value for each precision
            .add_field(sql="TIME '23:59:59'", python=time(23, 59, 59))
            .add_field(sql="TIME '23:59:59.9'", python=time(23, 59, 59, 900000))
            .add_field(sql="TIME '23:59:59.99'", python=time(23, 59, 59, 990000))
            .add_field(sql="TIME '23:59:59.999'", python=time(23, 59, 59, 999000))
            .add_field(sql="TIME '23:59:59.9999'", python=time(23, 59, 59, 999900))
            .add_field(sql="TIME '23:59:59.99999'", python=time(23, 59, 59, 999990))
            .add_field(sql="TIME '23:59:59.999999'", python=time(23, 59, 59, 999999))
            # round down
            .add_field(sql="TIME '00:00:00.0000001'", python=time(0, 0, 0))
            .add_field(sql="TIME '12:34:56.1234561'", python=time(12, 34, 56, 123456))
            .add_field(sql="TIME '23:59:59.9999994'", python=time(23, 59, 59, 999999))
            .add_field(sql="TIME '23:59:59.9999994999'", python=time(23, 59, 59, 999999))
            # round down, max value
            .add_field(sql="TIME '00:00:00.0000004'", python=time(0, 0, 0))
            .add_field(sql="TIME '00:00:00.00000049'", python=time(0, 0, 0))
            # round up, min value
            .add_field(sql="TIME '00:00:00.0000005'", python=time(0, 0, 0))
            .add_field(sql="TIME '00:00:00.00000050'", python=time(0, 0, 0))
            # round up, max value
            .add_field(sql="TIME '00:00:00.0000009'", python=time(0, 0, 0, 1))
            .add_field(sql="TIME '00:00:00.00000099'", python=time(0, 0, 0, 1))
            # round up to next day, min value
            .add_field(sql="TIME '23:59:59.9999995'", python=time(0, 0, 0))
            .add_field(sql="TIME '23:59:59.99999950'", python=time(0, 0, 0))
            # round up to next day, max value
            .add_field(sql="TIME '23:59:59.9999999'", python=time(0, 0, 0))
            .add_field(sql="TIME '23:59:59.99999999'", python=time(0, 0, 0))
    ).execute()
```

Reply:
I found another Trino issue: `SELECT TIME '23:59:59.99999999` returns `24:00:00.000`. Note also that this PR doesn't implement `PARAMETRIC_DATETIME` (so only up to millis precision).

See https://github.com/trinodb/trino/pull/15138

Reply:
So nice that you've managed to find so many bugs just working on this piece of code. 😄